feat: enabled integration with HA-valkey via network changes and documentation (#245)
defenseunicorns/uds-package-gitlab: docs/configuration.md

GitLab uses Redis as a key value store for caching, job queueing and more and supports external providers (such as Elasticache) as well as the [UDS Valkey](https://github.com/defenseunicorns/uds-package-valkey/) package to provide the service.

### Valkey HA Configuration

> The [Valkey UDS Application Package](https://github.com/defenseunicorns/uds-package-valkey) supports the HA replicated architecture ([as of v8.0.1-uds.1](https://github.com/defenseunicorns/uds-package-valkey/releases/tag/v8.0.1-uds.1)) where there is one write node (called a primary), multiple read nodes, and sentinels as side-cars who will elect a new primary in the event the existing primary goes down.

Reviewer: ```suggestion
The [Valkey UDS Package](https://github.com/defenseunicorns/uds-package-valkey) supports the HA replicated architecture ([as of v8.0.1-uds.1](https://github.com/defenseunicorns/uds-package-valkey/releases/tag/v8.0.1-uds.1)) where there is one write node (called a primary), multiple read nodes, and sentinels as side-cars who will elect a new primary in the event the existing primary goes down.
```

We usually don't use the word Application in Packages and omitting it is more terse and a little easier to read

> This configuration further [documented in the repo](https://github.com/defenseunicorns/uds-package-valkey/blob/main/docs/configuration.md#high-availability). All configuration changes required to connect an HA Valkey to GitLab will be performed at the _bundle_ level. To connect that HA Valkey to Gitlab:

Reviewer: ```suggestion
This configuration is further [documented in the Valkey repo](https://github.com/defenseunicorns/uds-package-valkey/blob/main/docs/configuration.md#high-availability). All configuration changes required to connect an HA Valkey to GitLab will be performed at the _bundle_ level. To connect the HA Valkey to Gitlab:
```

> 1. Perform the [configuration changes](https://github.com/defenseunicorns/uds-package-valkey/blob/main/docs/configuration.md#configuration-changes) to configure the Valkey Application Package to deploy an HA instance in your bundle.

Reviewer: ```suggestion
1. Perform the [configuration changes](https://github.com/defenseunicorns/uds-package-valkey/blob/main/docs/configuration.md#configuration-changes) to configure the Valkey Package to deploy an HA instance in your bundle.
```

> 2. **WARNING: what follows is unintuitive.** Change the `global.redis.host` value to be the _name_ of the primary node's role. By default, that is `mymaster`. This value is no longer to be the address for redis. This is somewhat intuitive considering that GitLab will be using the sentinel to find the redis address, but needs to know the name of the primary's role, so this value is still key info required in finding the redis host, but the value _is not_ the redis host address.

Reviewer: ```suggestion
2. Change the `global.redis.host` value to be the _name_ of the primary node's role. By default, that is `mymaster`. This value is no longer to be the address for redis.

> [!WARNING]
> This may seem unintuitive until you consider that GitLab will be using the sentinel to find the redis address, but needs to know the name of the primary's role.  This value is still key info required in finding the redis host, but the value ends up _not_ being the redis host address.
```

Reviewer: Generally we use admonitions to make callouts pop in GitHub

Author: I don't think I'd seen these admonitions before. Cool.

  ```yaml
  packages:
    - name: gitlab
      overrides:
        gitlab:
          gitlab:
            values:
              - path: global.redis.host
                value: mymaster
  ```

> 3. _At the bundle level_, override the `global.redis.sentinels` path in the GitLab chart with a list of the valkey sentinel headless addresses, shown below. This must be done _at the bundle level_ due to limitations with Zarf variables. While these excerpts show the full yaml path for clarity, in practice, this change would be right next to the prior change. They would also likely be best done through UDS variables, not hardcoded values, so the operator can scale Valkey and update GitLab without rebuilding the bundle.

Reviewer: I'm confused by this statement - there are no Zarf variables in Valkey to set so ofc they wouldn't work? Were there issues with _bundle_ `variables` vs `values`? (technically both bundle variables and bundle values are at the bundle level).

Author: I think it's confusing partly because I've got observations from two time periods stuck together here. I think the Zarf variable comment comes from an idea a few weeks ago that was stopped by the fact we can't pass dictionaries or lists via zarf variables, only strings.

The recommendation is to make the number of Valkey replicas and the list of sentinels into bundle variables so they can be modified at deploy time. Our customer especially is interested in us thinking of this sort of thing before-hand. I will think about how to write that to be clearer.

Author: Just cut everything after the first sentence. Ideal bundle setup is out of scope.
